packages/mobx-react-lite/__tests__/observer.test.tsx

```diff
@@ -1084,3 +1084,24 @@ test("Anonymous component displayName #3192", () => {
     expect(observerError.message).toEqual(memoError.message)
     consoleErrorSpy.mockRestore()
 })
+
+test("#3648 enableStaticRendering doesn't warn with observableRequiresReaction/computedRequiresReaction", () => {
+    consoleWarnMock = jest.spyOn(console, "warn").mockImplementation(() => {})
+    try {
+        enableStaticRendering(true)
+        mobx.configure({ observableRequiresReaction: true, computedRequiresReaction: true })
+        const o = mobx.observable.box(0, { name: "o" })
+        const c = mobx.computed(() => o.get(), { name: "c" })
+        const TestCmp = observer(() => <div>{o.get() + c.get()}</div>)
+
+        const { unmount, container } = render(<TestCmp />)
+        expect(container).toHaveTextContent("0")
+        unmount()
+
+        expect(consoleWarnMock).toMatchSnapshot()
```

Comment:
Q: what does this assertion do? Should that be `expect(consoleWarnMock).not.toBeCalled()`?

Reply:
The mocked function serializes to an object with an array of calls and results, eg like here:
https://github.com/mobxjs/mobx/blob/b05595be8614fa8ba2137fd4050f933bd835a044/packages/mobx-react/__tests__/__snapshots__/observer.test.tsx.snap#L41-L55
So since it's serialized only to `[MockFunction]` it means it wasn't called.

Reply:
Got ya! thanks